activejob/lib/active_job/queue_adapter.rb

```diff
@@ -1,32 +1,41 @@
 require 'active_job/queue_adapters/inline_adapter'
+require 'active_support/core_ext/class/attribute'
 require 'active_support/core_ext/string/inflections'
 
 module ActiveJob
   # The <tt>ActiveJob::QueueAdapter</tt> module is used to load the
   # correct adapter. The default queue adapter is the :inline queue.
   module QueueAdapter #:nodoc:
-    extend ActiveSupport::Concern
+    # We can't use ActiveSupport::Concern here because we need to `prepend`
+    # ClassMethods rather than `extend`ing it.
+    def self.included(klass)
+      klass.class_attribute :queue_adapter, instance_accessor: false, instance_predicate: false
+      klass.singleton_class.prepend(ClassMethods)
```

Comment:
yeah, this is similar to what I had before applying the second commit on this branch at @cristianbica's request. Would you like me to strip that commit off?